report/metrics.go

```diff
@@ -0,0 +1,268 @@
+package report
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"time"
+
+	"github.com/mndrix/ps"
+)
+
+// Metrics is a string->metric map.
+type Metrics map[string]Metric
```

Comment:
Okay I tried that and couldn't make it work nicely.  Lets leave for now, and I'll make this immutable in the future.